2025/kirtanchandak/kirtanchandak-gsoc-2025-report.md

```diff
@@ -0,0 +1,224 @@
+
+- **Issue:** https://github.com/openfoodfacts/folksonomy_api/issues/19
+- **PR:** https://github.com/openfoodfacts/folksonomy_api/pull/263
+
+#### **Add Data to Local Database**
+
+Added support for inserting data directly into the local folksonomy database. This makes it easier for contributors to test changes and add sample data without affecting the production environment.
+
+- **Slack Discussion:** https://openfoodfacts.slack.com/archives/C01M1T2E8L8/p1741101888486549
+- **PR:** https://github.com/openfoodfacts/folksonomy_api/pull/252
+
+#### **Enable Getting Multiple Properties at Once**
+
+Enhanced the folksonomy API to allow fetching more than one property in a single request.
+
+- **Issue:** https://github.com/openfoodfacts/folksonomy_api/issues/236
+- **PR:** https://github.com/openfoodfacts/folksonomy_api/pull/244
+
+#### **Add Typing for OpenAPI Response for `/keys`**
+
+Improved the folksonomy API's OpenAPI documentation by adding proper typing for the `/keys` endpoint response.
+
+- **Issue:** https://github.com/openfoodfacts/folksonomy_api/issues/149
+- **PR:** https://github.com/openfoodfacts/folksonomy_api/pull/230
+
+## Feature Work Delivered For Google Summer of Code 2025
+
+### **Folksonomy Editor Integration**
+
+Integrated existing `folksonomy-editor` webcomponent from the openfoodfacts-webcomponents into OpenFoodFacts. This work involved ensuring compatibility, fixing integration bugs, and making sure the components work seamlessly within the folksonomy interface. The integration improves code reusability and maintains consistency across different Open Food Facts applications.
+
+<p align="center">
+  <img src="./assets/folksonomy-editor.png" width="700" />
+</p>
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/openfoodfacts-server/pull/12136
+
+### **Mobile Suggestion System**
+
+Implemented a comprehensive suggestion system that works effectively on mobile devices. The system uses the folksonomy API endpoints, particularly the `/values/{k}` endpoint.
+
+Documentation - https://api.folksonomy.openfoodfacts.org/docs#/Keys%20%26%20Values/get_unique_values_values__k__get
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/folksonomy_api/pull/263
+
+### **Property Generation Wizard**
+
+Created a simple yet powerful property generation wizard that automatically creates corresponding wiki pages for new properties. This tool streamlines the process of adding new property types to the system and ensures proper documentation is maintained.
+
+The wizard guides users through the process of defining new properties, validates the input, and automatically generates the necessary wiki documentation, making it easier for the community to understand and use new properties.
+
+<p align="center">
+  <video width="700" controls>
+    <source src="./assets/folksonomy-wizard.mp4" type="video/mp4">
+    Your browser does not support the video tag.
+  </video>
+</p>
```

Comment:
Beware that it does not display at all on Github, so I would add also add a link to the video.
```suggestion
<p align="center">
  <video width="700" controls>
    <source src="./assets/folksonomy-wizard.mp4" type="video/mp4">
    Your browser does not support the video tag.
  </video>
  [video showing the wizard in action](./assets/folksonomy-wizard.mp4)
</p>
```